### ISSUE TITLE
[XLA] AlgebraicSimplifier folds a negative (cropping) pad into reduce_window / convolution window padding

### ISSUE THREAD

**Issue by EylonKrause** [openxla/xla#48715]
XLA's `AlgebraicSimplifier` folds a `kPad` whose edge padding is **negative** (a crop) into the
window padding of a consuming `reduce_window` or `convolution` by adding the two together. Cropping
and then padding the same side is not the same function as the net padding, so the fused program
uses input elements where padding values belong. Only the jit-compiled program is affected; the
op-by-op path is correct.

### Reproducer

```python
import os
os.environ["JAX_PLATFORMS"] = "cpu"
import numpy as np, jax, jax.numpy as jnp
from jax import lax

# --- site 1: pad -> reduce_window ---------------------------------------------------------
x = jnp.array([[669, -763]], dtype=jnp.int32)           # shape (1, 2)
cfg = [(0, 0, 1), (-1, 0, 0)]                            # dim0: interior 1 on a size-1 dim (no-op); dim1: crop x[:,0]

def f(a):
    p = lax.pad(a, jnp.int32(0), cfg)                    # -> [[-763]]
    return lax.reduce_window(p, jnp.int32(0), lax.add, (1, 2), (1, 1), ((0, 0), (1, 0)))  # [0,-763] window 2 -> [[-763]]

expected = np.array([[-763]])
print("reduce_window expected:", expected.tolist(), "eager:", np.asarray(f(x)).tolist(), "jit:", np.asarray(jax.jit(f)(x)).tolist())
print("  optimized HLO root:", [l.strip()[:90] for l in jax.jit(f).lower(x).compile().as_text().splitlines() if "reduce(" in l])

# --- site 2: pad -> conv (canonical NHWC/HWIO so no ConvCanonicalization transpose sits between them) ---
for dtype in (jnp.int32, jnp.float32):
    xc = jnp.array([5, 7, 11, 13], dtype=dtype).reshape(1, 1, 4, 1)     # NHWC: N=1 H=1 W=4 C=1
    kc = jnp.array([1, 100], dtype=dtype).reshape(1, 2, 1, 1)           # HWIO: KH=1 KW=2 I=1 O=1
    dn = lax.ConvDimensionNumbers((0, 3, 1, 2), (3, 2, 0, 1), (0, 3, 1, 2))
    cfgc = [(0, 0, 0), (0, 0, 1), (-1, 0, 0), (0, 0, 0)]                # H: no-op interior; W: crop first element

    def g(a, b):
        p = lax.pad(a, dtype(0), cfgc)                                  # W -> [7, 11, 13]
        return lax.conv_general_dilated(p, b, (1, 1), ((0, 0), (1, 0)), dimension_numbers=dn)  # W -> [0,7,11,13]

    exp_c = [0 * 1 + 7 * 100, 7 + 11 * 100, 11 + 13 * 100]              # [700, 1107, 1311]
    print(dtype.__name__, "conv expected:", exp_c, "eager:", np.asarray(g(xc, kc)).ravel().tolist(),
          "jit:", np.asarray(jax.jit(g)(xc, kc)).ravel().tolist())
    print("  optimized HLO conv:", [l.strip()[:100] for l in jax.jit(g).lower(xc, kc).compile().as_text().splitlines() if "convolution(" in l])
```

Output on my machine:

```
reduce_window expected: [[-763]]        eager: [[-763]]        jit: [[-94]]
  optimized HLO root: %reduce.1 = s32[1]{0} reduce(%param_0.1, %constant.0), dimensions={1}
int32 conv    expected: [700, 1107, 1311]  eager: [700, 1107, 1311]  jit: [705, 1107, 1311]
float32 conv  expected: [700, 1107, 1311]  eager: [700.0, ...]       jit: [705.0, 1107.0, 1311.0]
  optimized HLO conv: ROOT %convolution.2 = ... convolution(%param_0, %param_1), window={size=1x2}
```

### Analysis

In the `reduce_window` case the pad crops `x[:, 0]` away and the window padding then adds a zero, so
the reduction runs over `[0, -763]` and must give `-763`. The optimized HLO shows the pad gone and a
plain `reduce` over the *uncropped* input, giving `669 + (-763) = -94`. In the convolution case the
padded `W` axis must be `[0, 7, 11, 13]`, but the fused program uses `x0`, giving `5 + 7*100 = 705`
in the first output element.

The folds are `HandleReduceWindow`'s
`window_dim.set_padding_low(padding_low + base_dilation * edge_padding_low)` and `FoldConvInputPad`'s
`w.set_padding_low(w.padding_low() + p.edge_padding_low())`
(`xla/hlo/transforms/simplifiers/algebraic_simplifier.cc`). Both assume the pad's edge padding is
non-negative. They become reachable in the same pass because `HandlePad` strips a no-op interior pad
on a size-1 dimension and returns before its own negative-padding-to-slice rewrite, so the consumer
still sees a live negative pad.

`lax.pad` with negative edge padding is documented and JAX generates it, so guarding the folds on
`edge_padding_low >= 0` (and likewise for the high side) would be enough.

### Environment

- `jax` 0.11.1, `jaxlib` 0.11.1 (pip), CPU backend, `JAX_PLATFORMS=cpu`
- Linux x86-64 (AVX-512), Python 3.12

### Notes

Found by a differential fuzzer comparing XLA:CPU against a NumPy oracle written from the operation semantics (validated by hand and against `jax._src.lax_reference`), then minimised. I am filing this as an issue rather than a pull request, per the repository's contribution guidelines: this report was prepared with the help of an AI coding assistant (Claude), and I reproduced every result above myself on the environment listed.

**Comment by penumallas1261** [comment]
Internal reproducer submitted in cl/985017811.
